packages/grid/_modules_/grid/components/columnHeaders/GridColumnHeaderSortIcon.tsx

```diff
@@ -30,24 +31,35 @@ const useUtilityClasses = (ownerState: OwnerState) => {
   return composeClasses(slots, getDataGridUtilityClass, classes);
 };
 
-function getIcon(icons: GridIconSlotsComponent, direction: GridSortDirection, className: string) {
-  let Icon = icons.ColumnUnsortedIcon;
+function getIcon(
+  icons: GridIconSlotsComponent,
+  direction: GridSortDirection,
+  className: string,
+  sortingOrder: GridSortDirection[],
+) {
+  let Icon;
   if (direction === 'asc') {
     Icon = icons.ColumnSortedAscendingIcon;
   } else if (direction === 'desc') {
     Icon = icons.ColumnSortedDescendingIcon;
+  } else {
+    Icon = icons.ColumnUnsortedIcon;
+    const [nextSortDirection] = sortingOrder;
+    return Icon ? (
+      <Icon fontSize="small" className={className} data-nextsortdirection={nextSortDirection} />
```

Comment:
Do you need `data-nextsortdirection` for your application? Is not `aria-sort` already sufficient? I'm asking because the logic could be moved to `GridColumnHeaderSortIcon` if the attribute is only used to decide which icon to display.

Reply:
I think this is only used for the mouse-over hint for the next sort direction when the current state is unsorted. I couldn't see an obvious way to contain the logic to `GridColumnHeaderSortIcon` while still returning `GridColumnUnsortedIcon` for the unsorted state.

Good point about `aria-sort` - I've switched to that instead of the custom data attribute.

Reply:
`aria-sort` can't be used twice. 

From https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-sort

> If a grid or table provides sort functionality, the aria-sort attribute should be set to either ascending or descending (or other) on the header cell element for the sorted column or row.

I pushed a small change to pass the `sortingOrder` prop directly to the icon, no data attribute. Are you OK with this approach?

Reply:
Thanks - looks great and is what I'd intended to do in https://github.com/mui-org/material-ui-x/pull/3449/commits/b94f3bf977b2eb2a705bc111cb801f01999c0cab (still confused what the difference between that and your working https://github.com/mui-org/material-ui-x/pull/3449/commits/a3b11f1a793aeda8598077ec14621b4f2a2811c6 version is).

Reply:
In b94f3bf you were using the `sortingOrder` prop and also spreading it into the DOM element, that's reason for the "React does not recognize the `sortingOrder` prop on a DOM element." warning. In a3b11f1, I don't spread it into the DOM node.